app/components/Base/AnimatedFox.js

```diff
@@ -0,0 +1,1574 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { WebView } from '@metamask/react-native-webview';
+import { gyroscope } from 'react-native-sensors';
+
+function round(value, decimals) {
+  return Number(Math.round(value + 'e' + decimals) + 'e-' + decimals);
+}
+
+class AnimatedFox extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.position = {
+      beta: 0,
+      gamma: 0,
+    };
+
+    this.subscription = gyroscope.subscribe(
+      ({ x, y }) => {
+        this.position = {
+          beta: this.position.beta - round(x * -10, 4),
+          gamma: this.position.gamma - round(y * -10, 4),
+        };
+
+        requestAnimationFrame(() => {
+          const JS = `
+					(function () {
+						const event = new CustomEvent('nativedeviceorientation', {
+							detail: {
+								alpha:${this.position.alpha},
+								beta:${this.position.beta},
+								gamma:${this.position.gamma}
+							}
+						});
+
+						window.dispatchEvent(event);
+					})();
+				`;
+          const { current } = this.webview;
+          current && current.injectJavaScript(JS);
+        });
+      },
+      () => {
+        //Nothing to do here, sensor not available
+      },
+    );
+  }
+
+  componentWillUnmount() {
+    this.subscription.unsubscribe();
+  }
+
+  webview = React.createRef();
+
+  render() {
+    return (
+      <WebView
+        ref={this.webview}
+        style={{ flex: 1 }}
```

Comment:
```suggestion
        // eslint-disable-next-line react-native/no-inline-styles
        style={{ flex: 1 }}
```
I guess the right thing to do is to add a proper style declaration, though.

Reply:
Added a new file with the component all refactored: https://github.com/MetaMask/metamask-mobile/pull/9695/commits/070ebc43e366f9633398b7c88fa2257dfb2c3aa7